From ade680aa92feab68eb06424ad416dfbb58d9bc05 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Mon, 27 Jun 2011 14:33:11 +0200
Subject: [PATCH] serial: extend rs485 support with full/half duplex selection

We create a new RS485 configuration flag SER_RS485_FULL_DUPLEX. When
set, the RS485 link must be configured in full duplex mode by the
serial driver. When unset, the RS485 link must be configured in half
duplex mode by the serial driver.

This allows userspace applications to adjust the full/half duplex
selection using the existing TIOCGRS485 and TIOCSRS485 ioctl calls.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 include/linux/serial.h |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/include/linux/serial.h b/include/linux/serial.h
index ef91406..8a4d294 100644
--- a/include/linux/serial.h
+++ b/include/linux/serial.h
@@ -211,6 +211,7 @@ struct serial_rs485 {
 #define SER_RS485_RTS_ON_SEND		(1 << 1)
 #define SER_RS485_RTS_AFTER_SEND	(1 << 2)
 #define SER_RS485_RTS_BEFORE_SEND	(1 << 3)
+#define SER_RS485_FULL_DUPLEX		(1 << 4)
 	__u32	delay_rts_before_send;	/* Milliseconds */
 	__u32	delay_rts_after_send;	/* Milliseconds */
 	__u32	padding[5];		/* Memory is cheap, new structs
-- 
1.7.4.1

